File: Lesson9/app/src/main/java/ru/mirea/vasilev/lesson9/presentation/MainActivity.java
package ru.mirea.vasilev.lesson9.presentation;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;


import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import ru.mirea.vasilev.domain.models.Movie;
import ru.mirea.vasilev.lesson9.R;

public class MainActivity extends AppCompatActivity {

    private MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewModel = new ViewModelProvider(this, new ViewModelFactory(this)).get(MainViewModel.class);

        EditText edittext = findViewById(R.id.editTextMovie);
        TextView textView = findViewById(R.id.textViewMovie);

        viewModel.getFavoriteMovie().observe(this, new Observer<String>() {
            @Override
            public void onChanged(String s) {
                textView.setText(s);
            }
        });

//        MovieStorage sharedPrefMovieStorage = new SharedPrefMovieStorage(this);
//        MovieRepository movieRepository = new MovieRepositoryImpl(sharedPrefMovieStorage);

        findViewById(R.id.buttonSaveMovie).setOnClickListener(v -> {
            viewModel.SaveMovie(new Movie(2, edittext.getText().toString()));
//            textView.setText(String.format("Save result %s", result));
        });

        findViewById(R.id.buttonGetMovie).setOnClickListener(v -> {
            viewModel.getMovie();
//            textView.setText(movieName);
        });
    }
}

File: lesson9/app/src/main/java/ru/mirea/vasilevmn/lesson9/presentation/MainActivity.kt
package ru.mirea.vasilevmn.lesson9.presentation

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import ru.mirea.vasilevmn.lesson9.R
import ru.mirea.vasilevmn.lesson9.data.repository.MovieRepositoryImpl
import ru.mirea.vasilevmn.lesson9.domain.models.Movie
import ru.mirea.vasilevmn.lesson9.domain.repository.MovieRepository
import ru.mirea.vasilevmn.lesson9.domain.usecases.GetFavoriteFilmUseCase
import ru.mirea.vasilevmn.lesson9.domain.usecases.SaveMovieToFavoriteUseCase


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val text = findViewById<EditText>(R.id.editTextMovie)
        val textView = findViewById<TextView>(R.id.textViewMovie)
        val movieRepository: MovieRepository = MovieRepositoryImpl(this)

        findViewById<View>(R.id.buttonSaveMovie).setOnClickListener {
            val result = SaveMovieToFavoriteUseCase(movieRepository).execute(
                Movie(2, text.text.toString())
            )
            textView.text = String.format("Save result %s", result)
        }

        findViewById<View>(R.id.buttonGetMovie).setOnClickListener {
            val movie: Movie? = GetFavoriteFilmUseCase(movieRepository).execute()
            textView.text = java.lang.String.format("Save result %s", movie?.name)
        }
    }
}